clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -23333,54 +23333,67 @@ llvm::CallInst *CodeGenFunction::MaybeEmitFPBuiltinofFD(
       // attribute.
       return nullptr;
     case Builtin::BItan:
+    case Builtin::BItanf:
       FPAccuracyIntrinsicID = Intrinsic::fpbuiltin_tan;
       break;
     case Builtin::BItanh:
+    case Builtin::BItanhf:
       FPAccuracyIntrinsicID = Intrinsic::fpbuiltin_tanh;
       break;
     case Builtin::BIlog2:
       FPAccuracyIntrinsicID = Intrinsic::fpbuiltin_log2;
       break;
     case Builtin::BIlog1p:
+    case Builtin::BIlog1pf:
       FPAccuracyIntrinsicID = Intrinsic::fpbuiltin_log1p;
       break;
     case Builtin::BIcos:
```

Comment:
For some of the functions where we didn't need 32-bit versions, we might not need the calls to be handled here at all (which I think is the case with cos). For others, like ldexp and hypot, we probably will need them, but my tests just weren't hitting those cases. I don't see a downside to adding handlers for them now. Sorry if I missed this context when answering your question via IM.